configuration/pih/scripts/global/zl.js

```diff
@@ -185,32 +185,35 @@ function setUpObsWithObsDateTime(widgetId){
     }
 }
 
-function returnVisitDateValidation(encounterDate,returnVisitDateLessThanEncounterDateMsg,badReturnVisitDateMsg){
+function setupReturnVisitDateValidation(encounterDate,returnVisitDateLessThanEncounterDateMsg,badReturnVisitDateMsg){
 
-   const domEl= jq("#apptDate").find("input[type=text]");
+   const domEl= getField('apptDate.value');
    const yrRange = encounterDate.getFullYear() + ":" + (new Date().getFullYear() + 1);
-   domEl.datepicker('option','yearRange', yrRange); 
-   domEl.prop("readonly", "readonly"); 
-
-  jq(domEl).change(function(){
-    const nextVisitDate = domEl.datepicker('getDate');
-    const months = (nextVisitDate.getFullYear() - encounterDate.getFullYear()) * 12;
-
-    if(months < 12  && nextVisitDate > encounterDate){
-      jq("#return-visit-date-error-message").text('')
-      jq('.submit').prop("disabled", false);
-    }
-    else if(months == 12 && nextVisitDate.getMonth() <= encounterDate.getMonth()){
-      jq("#return-visit-date-error-message").text('')
-      jq('.submit').prop("disabled", false);
-    }else if(months <= 0 && encounterDate > nextVisitDate){
-        jq("#return-visit-date-error-message").text(returnVisitDateLessThanEncounterDateMsg)
-        jq('.submit').prop("disabled", true);
-    }
-    else{
-      jq("#return-visit-date-error-message").text(badReturnVisitDateMsg)
-      jq('.submit').prop("disabled", true);
-    }
-
-  })        
+   domEl.datepicker('option','yearRange', yrRange);
+   domEl.prop("readonly", "readonly");
+
+   const returnVisitDateValidation = function() {
+     const nextVisitDate = domEl.datepicker('getDate');
+     const months = (nextVisitDate.getFullYear() - encounterDate.getFullYear()) * 12;
+
+     if (months < 12  && nextVisitDate > encounterDate) {
+       getField('apptDate.error').text('').hide();
+       return true;
```

Comment:
Note that I replaced the code that disables/enabled the submit with one that returns true or false.  I'll explain the true/false below, but the main problem with disabling the submit here is: what if there's another validatoin function on this page? If a user enters an invalid value for appointment date, but then enters some valid data for the other field, the submit button will be re-enabled even though the appt date is still invalid.

We've added some custom functions to HTML Form Entry to help us get around this problem, see below.